### ISSUE TITLE
Signs do not show up at all and no hunks are detected

### ISSUE THREAD

**Issue by rkoval** [airblade/vim-gitgutter#522]
> What is the latest commit SHA in your installed vim-gitgutter?

a7206c41fc174f576b3f7085d48319dbdad760a5

> What vim/nvim version are you on?

```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Apr  4 2018 11:14:23)
macOS version
Included patches: 1-1633
```
... but also broken on
```
NVIM v0.2.2
Build type: Release
LuaJIT 2.0.5
```

> If no signs are showing at all, what does `:echo b:gitgutter.path` give?

Name of the current file

> If no signs are showing at all, and the `path` value is a path and not `-2`, does it work with `let g:gitgutter_grep=''`?

Yes

---
`:GitGutterDebug` output:
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Apr  4 2018 11:14:23)
macOS version
Included patches: 1-1633

git version 2.17.0

grep (BSD grep) 2.5.1-FreeBSD
usage: grep [-abcDEFGHhIiJLlmnOoqRSsUVvwxZ] [-A num] [-B num] [-C[num]]	[-e pattern] [-f file] [--binary-files=value] [--color=when]	[--context[=num]] [--directories=action] [--label] [--line-buffered]	[--null] [pattern] [file ...]

updatetime=4000
shell=/usr/local/bin/zsh
shellcmdflag=-c
shellpipe=2>&1| tee
shellquote=
shellredir=>%s 2>&1
shellslash [n/a]
shelltemp=1
shelltype [n/a]
shellxescape=
shellxquote=
```

---

vim-gitgutter stopped working suddenly, but I'm not sure what would have caused it to do so. I've attempted the [troubleshooting tips](https://github.com/airblade/vim-gitgutter/blob/master/README.mkd#troubleshooting), but to no avail. I've also tried disabling all of my plugins _except_ vim-gitgutter, but the same behavior persists (no signs and no hunks detected on a tracked file with unstaged changes). The only thing that I can maybe think of is that maybe it's not compatible with the most recent version of git that I'm using? I wasn't able to find a compatibility list anywhere here though, so I'm not sure.

Please let me know if there's anything else I can provide to help troubleshoot...

**Comment by airblade** [comment]
Thanks for all the detail.  It looks the problem is with grep, since the signs show up when you tell the plugin not to use grep.

Do you have `grep` aliased to anything else, or any grep options that would change its output from plain text without colours?

**Comment by airblade** [comment]
Oh, ok.  Good to know that grep is not the problem.

Since `b:gitgutter_path` shows the path to the file, the plugin is correctly establishing that the file can and should be processed.

Does `let g:gitgutter_async=0` make any difference?

Do you use tmux?  Sometimes that gets configured to put junk in its output (e.g. #323).

Please could you switch on logging with `let g:gitgutter_log=1`, force a couple of runs with `:GitGutter`, and post the logfile you'll find where gitgutter is installed?

**Comment by rkoval** [comment]
> Does let g:gitgutter_async=0 make any difference?

Not that I can tell.

> Do you use tmux? Sometimes that gets configured to put junk in its output (e.g. #323).

My main setup uses MacVim, but I've tried using Neovim in a non-tmux terminal--both are not working.

> Please could you switch on logging with let g:gitgutter_log=1, force a couple of runs with :GitGutter, and post the logfile you'll find where gitgutter is installed?

I've set this option, but I can't actually find any file named gitgutter.log in my plugin's installation directory or any of its subdirectories...? Here's hopefully a helpful video of my attempt to get it working: https://cl.ly/020R17420e1V

**Comment by airblade** [comment]
Thanks for the screencast and how odd – I've never known the log file to not show up!

The plugin places the log file like this (in `autoload/gitgutter/debug.vim`):

```viml
let s:plugin_dir  = expand('<sfile>:p:h:h:h').'/'
let s:log_file    = s:plugin_dir.'gitgutter.log'
```

I think `expand()` does not resolve symlinks, so if your plugin's installation dir is itself a symlink, maybe the log file will end up at the symlink source?

I noticed in your config a [couple of variables](https://github.com/rkoval/dotfiles/blob/80b35ea53c82c6bbd71b2fffd1465b1b193faa19/vim/config/gitgutter.vimrc#L2-L3) that the plugin doesn't use any more, which you can delete.

**Comment by rkoval** [comment]
Ah, yeah, hardcoding the gitgutter.log file to a path I know is not behind a symlink causes it to appear. See below:


```
==== start log session ====

  0.000027 function gitgutter#init_buffer[5]..gitgutter#utility#set_repo_path[11]..gitgutter#async#execute[1]:
  0.000027 [async] cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && git ls-files --error-unmatch --full-name -- README.mkd

  0.037980 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[67]..gitgutter#async#execute[1]:
  0.037980 [async] cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/7.1.mkd | grep '^@@ ' || exit 0)

  0.777930 function <SNR>240_on_exit_nvim[2]..gitgutter#diff#handler[1]:
  0.777930 

  3.342523 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[67]..gitgutter#async#execute[1]:
  3.342523 [async] cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/7.1.mkd | grep '^@@ ' || exit 0)

  3.369111 function <SNR>240_on_exit_nvim[2]..gitgutter#diff#handler[1]:
  3.369111 

 63.441447 function gitgutter#all[5]..gitgutter#process_buffer[8]..gitgutter#diff#run_diff[67]..gitgutter#async#execute[1]:
 63.441447 [async] cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/7.1.mkd | grep '^@@ ' || exit 0)

 63.477018 function <SNR>240_on_exit_nvim[2]..gitgutter#diff#handler[1]:
 63.477018 

 65.195381 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[67]..gitgutter#async#execute[1]:
 65.195381 [async] cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/7.1.mkd | grep '^@@ ' || exit 0)

 65.221710 function <SNR>240_on_exit_nvim[2]..gitgutter#diff#handler[1]:
 65.221710 

 66.666270 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[67]..gitgutter#async#execute[1]:
 66.666270 [async] cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/7.1.mkd | grep '^@@ ' || exit 0)

 66.693128 function <SNR>240_on_exit_nvim[2]..gitgutter#diff#handler[1]:
 66.693128 

 67.446574 function gitgutter#all[5]..gitgutter#process_buffer[8]..gitgutter#diff#run_diff[67]..gitgutter#async#execute[1]:
 67.446574 [async] cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvimYS1tNw/7.1.mkd | grep '^@@ ' || exit 0)

 67.484966 function <SNR>240_on_exit_nvim[2]..gitgutter#diff#handler[1]:
 67.484966 
```

Also, I did a `git bisect` on this repo, and I am able to confirm that my gitgutter stopped working at 5bfe5b92099e0d332a7ca26c0263fab509ee9617. It looks like there was a lot of changes made here, but this should definitely help us narrow down what the issue is.

Also also, I removed the unused config variables from my config—thanks for the tip!

**Comment by rkoval** [comment]
```sh
~/.v/d/r/g/a/vim-gitgutter » cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > a.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- a.mkd README.mkd | grep '^@@ ' || exit 0)

@@ -2,0 +3,14 @@
```
Does this help?

**Comment by airblade** [comment]
The command seems to work from the command line.

For some reason when nvim runs it, the output gets lost – both when running synchronously via `system()` or asynchronously via a job (since turning off async didn't help).

I'm not sure what the problem is.

Please could you turn off async and get some more log output?  I think it might be easier to see what's going on running synchronously.

**Comment by rkoval** [comment]
nvim log output with `let g:gitgutter_log=1`:

```
==== start log session ====

  0.000031 function gitgutter#init_buffer[5]..gitgutter#utility#set_repo_path[29]..gitgutter#utility#system[1]:
  0.000031 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && git ls-files --error-unmatch --full-name -- README.mkd

  0.038146 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
  0.038146 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/7.1.mkd | grep '^@@ ' || exit 0)

  0.068724 function gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
  0.068724 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd' is outside repository

  3.353434 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
  3.353434 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/7.1.mkd | grep '^@@ ' || exit 0)

  3.387034 function gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
  3.387034 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd' is outside repository

  9.138348 function gitgutter#all[5]..gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
  9.138348 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/7.1.mkd | grep '^@@ ' || exit 0)

  9.176410 function gitgutter#all[5]..gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
  9.176410 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd' is outside repository

 10.679847 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
 10.679847 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/7.1.mkd | grep '^@@ ' || exit 0)

 10.708522 function gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
 10.708522 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd' is outside repository

 18.431387 function gitgutter#all[5]..gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
 18.431387 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/7.1.mkd | grep '^@@ ' || exit 0)

 18.466568 function gitgutter#all[5]..gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
 18.466568 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/nvim4T4kC1/6.1.mkd' is outside repository
```

MacVim (with vim 8) log output with `let g:gitgutter_log=1`:

```


==== start log session ====

  0.000018 function gitgutter#init_buffer[5]..gitgutter#utility#set_repo_path[29]..gitgutter#utility#system[1]:
  0.000018 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && git ls-files --error-unmatch --full-name -- README.mkd

  0.119368 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
  0.119368 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/16.1.mkd | grep '^@@ ' || exit 0)

  0.173838 function gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
  0.173838 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd' is outside repository

  6.488294 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
  6.488294 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/16.1.mkd | grep '^@@ ' || exit 0)

  6.550662 function gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
  6.550662 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd' is outside repository

  8.896939 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
  8.896939 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/16.1.mkd | grep '^@@ ' || exit 0)

  8.958648 function gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
  8.958648 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd' is outside repository

 14.982564 function gitgutter#all[5]..gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
 14.982564 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/16.1.mkd | grep '^@@ ' || exit 0)

 15.039392 function gitgutter#all[5]..gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
 15.039392 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd' is outside repository

 17.129125 function gitgutter#process_buffer[8]..gitgutter#diff#run_diff[74]..gitgutter#utility#system[1]:
 17.129125 cd /Users/ryankoval/dotfiles/vim/dein/repos/github.com/airblade/vim-gitgutter && (git --no-pager show :README.mkd > /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd && git --no-pager -c "diff.autorefreshindex=0" -c "diff.noprefix=false" -c "core.safecrlf=false" diff --no-ext-diff --no-color -U0 HEAD -- /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/16.1.mkd | grep '^@@ ' || exit 0)

 17.188097 function gitgutter#process_buffer[17]..gitgutter#diff#handler[1]:
 17.188097 fatal: /var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd: '/var/folders/ln/ywkznnhj5nx_vwwgmqnc8ggr0000gn/T/vPe6a2x/15.1.mkd' is outside repository
```

Output from both neovim and MacVim (vim 8) both are pretty identical, but at least now it looks like we're getting somewhere. It looks like vim-gitgutter is running the diff against a temp version of the file vs. the file itself?

**Comment by airblade** [comment]
Excellent, finally some error messages :)

vim-gitgutter writes the buffer to a temporary file, and writes the staged version of the file to another temporary file, and diffs them.  This works because git-diff explicitly supports diffing files outside the repo.  I'm not sure why it's complaining in this case.

As a wild guess I wonder if it's to do with your symlinks?

**Comment by airblade** [comment]
Ah, can you try without setting `let g:gitgutter_diff_base='HEAD'`?

**Comment by rkoval** [comment]
I don't think it has to do with symlinks. I get the same exact same behavior/logs for files that are not behind any symlinks

**Comment by rkoval** [comment]
Oooh I think we found the problem/solution!

I was using `let g:gitgutter_diff_args = 'HEAD'` (`args`, not `base`), which I'm guessing I pulled somewhere from the vim doc at some point a while back (I haven't really changed this config in years). This config's use must have changed after the refactor. When I use your updated config `let g:gitgutter_diff_base='HEAD'` instead of what I had, vim-gitgutter starts working again _and_ tracks changes from the index!

I'll let you know if run into anything else with this, but I think we can safely call this issue fixed.

Thanks for your time and patience!
